department-of-veterans-affairs/va.gov-cms
# VACMS-10356 Add content validator for Preview URL links (#10356)

## Description

Closes https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10356.

## Testing done

- Added unit tests for validator
- Tested functionality locally

## Screenshots


## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

As user with ability to edit content:
1. Edit any node/page with rich text field, e.g. the VAMC system service description of a System Health Service.
2. Insert a link to a preview URL, e.g. `<a href="https://preview-staging.vfs.va.gov/example/path">Example Link!</a>`
3. Try to save the document and be blocked by errors and validate that:
   - [x] An error appears at the top of the page:
   <img width="485" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/ffdcef63-a18a-4a6f-b18a-98f0251c0c4a">

   - [x] The field title is highlighted red to indicate error:
   <img width="707" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/3eb165ce-5626-488d-9a7e-09ae1ccd4b0e">

   - [x] An error appears at the bottom of the field:
   <img width="853" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/c6aa98aa-80d9-48f9-90d3-daae26e114c9">

4. Edit any node/item with plain text field, e.g. description of a media image.
5. Add a link to a preview URL, e.g. `https://preview-staging.vfs.va.gov/example/path`
6. Try to save the document and be blocked by errors and validate that:
   - [x] An error appears at the top of the page:
   <img width="320" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/308cb38a-e6f6-4e44-aed7-3bee5dc5d0eb">

   - [x] The field title is highlighted red to indicate error:
   <img width="948" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/7217b394-d6e3-4152-bae3-966458c56370"> 

   - [x] An error appears at the bottom of the field:
   <img width="564" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/2030323/f3ddc2ad-f091-4ae0-b2bf-1c736d416a64">

7. Then validate Acceptance Criteria from issue
   - [x]  A link created to the preview server url will fail form validation and prevent save.
   - [x] Validator is covered by unit tests.

### Definition of Done


### Select Team for PR review
